solr/core/src/java/org/apache/solr/update/processor/AtomicUpdateDocumentMerger.java

```diff
@@ -69,10 +69,12 @@ public class AtomicUpdateDocumentMerger {
 
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+  protected final SolrQueryRequest request;
   protected final IndexSchema schema;
   protected final SchemaField idField;
 
   public AtomicUpdateDocumentMerger(SolrQueryRequest queryReq) {
+    request = queryReq;
```

Comment:
Maybe no need to retain the full request?
```suggestion
    skipUpdateIfMissing = queryReq.getParams().getBool("skipUpdateIfMissing", false);
```